### PR TITLE
some type annotations

### PR DESCRIPTION
**Pull request by peterbecich**
I am investigating a possible issue in `FileMonitor`. ~This PR adds a few type annotations. It also replaces `$` with `.` in some places.~

I have only tested the change by building Cabal, and the Github Actions.

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by jneira** [comment]
Could appveyor build be restarted? It belongs to @23Skidoo account i dont have rights to trigger the rebuild.

**Comment by Mikolaj** [comment]
Me neither. @emilypi, @fgaz? An option is to remove appveyor altogether --- it's having fits every once in a while.

**Review by jneira — APPROVED** [review]
Not sure if replacing `$` with `.` will be consistent with the rest of the codebase, but i am not against.
Add type annotations always is good and help new readers to understand the code, thanks!

**Comment by fgaz** [comment]
+1 for the annotations, but I'd avoid the reformatting and changing `$` with `.` unless there's some advantage in using the latter

**Review by jneira — APPROVED** [review]
LGTM, thanks for help readers of cabal codebase (specially to beginners)

**Comment by jneira** [comment]
@fgaz is this in good shape to merge now?

File: cabal-install/src/Distribution/Client/FileMonitor.hs
Comment (fgaz) [inline cabal-install/src/Distribution/Client/FileMonitor.hs:284]: this one

File: cabal-install/src/Distribution/Client/ProjectOrchestration.hs
Comment (fgaz) [inline cabal-install/src/Distribution/Client/ProjectOrchestration.hs:718]: and this

**Review by fgaz — COMMENTED** [review]
@jneira like I wrote above I'd avoid the reformatting. Finally, this should be squashed before merging (or just use the squash&merge button)

**Review by emilypi — APPROVED** [review]
Nice little quality of life improvement :)

**Comment by peterbecich** [comment]
@emilypi , yes, please merge. Thank you

**Pull request merged**
